cloudformation/slack-router-lambda.js

```diff
@@ -0,0 +1,104 @@
+const AWS = require('aws-sdk')
+const crypto = require('crypto')
+
+const region = process.env.AWS_REGION
+const accountId = process.env.AWS_ACCOUNT_ID
+
+let ssmConnection
+function getSSMConnection() {
+  if (ssmConnection) return ssmConnection
+
+  ssmConnection = new AWS.SSM()
+  return ssmConnection
+}
+
+let snsConnection
+function getSNSConnection() {
+  if (snsConnection) return ssmConnection
+
+  snsConnection = new AWS.SNS()
+  return snsConnection
+}
+
+function parseBody(event) {
+  const eventJSON = JSON.stringify(event.body, null, 2)
+  const eventBody = Buffer.from(eventJSON, 'base64').toString('ascii')
+  const bodyArray = eventBody.split('&')
+
+  const bodyObject = bodyArray.reduce((accumulator, currentValue) => {
+    const [key, value] = currentValue.split('=')
+
+    accumulator[key] = value
+
+    return accumulator
+  }, {})
+
+  return bodyObject
+}
+
+async function verifyRequest(req) {
+  const ssmParams = {
+    Name: 'slack-router-signing-secret',
+    WithDecryption: true,
+  }
+
+  const ssm = getSSMConnection()
+  const ssmResult = await ssm.getParameter(ssmParams).promise()
+  const slackSecret = ssmResult.Parameter.Value
+
```

Comment:
You can pass the SSM parameter directly as an environment variable to the lambda using `cf.ref`. Take a look at https://stackoverflow.com/questions/56027070/is-it-possible-to-use-ssm-parameters-in-environment-variables-for-a-lambda for an example.

Reply:
You can't assign an SSM Secure String as a Lambda environment variable. You get an error similar to this: `Reason: SSM Secure reference is not supported in: [AWS::Lambda::Function/Properties/Environment/Variables/VAR_NAME]`

Having the Lambda access the SSM parameter that it needs (and only what it needs) also means there will be no risk of CloudFormation logging that secret and follows the principle of least privelege. This way as well, if ever the SSM parameter is updated, there will be no need to update the stack since Lambda will always fetch the latest version of the parameter.

Reply:
Yeah agree with Andria here to fetch parameter store directly - @arunasank - is there a definite reason we should be accessing the values as env variables?